src/components/settings/Calendars.js

```diff
@@ -203,14 +219,61 @@ export default class Calendars extends AddDeleteSetting {
     this.setState({ hexColor })
   }
 
+  handleFieInputChange = event => {
+    const { verifyNewCalendar } = this.props
+
+    console.log('selected files', event.target.files[0])
+    if (event.target.files[0]) {
+      let file = event.target.files[0]
+      let fileExt = file.name.split('.').pop()
+
+      if (fileExt.toLowerCase() !== 'ics') {
+        alert('Invalid file type')
+        return
+      }
+
+      let fileReader = new FileReader()
+      fileReader.onloadend = e => {
+        console.log('read file complete')
+
+        let calendar = {
+          uid: uuid(),
+          type: 'ics',
+          name: this.state.calendarName,
+          hexColor: this.state.hexColor,
+          mode: 'read-only',
+          data: {
+            src: fileReader.result,
```

Comment:
`src` is used to contain a url. 

So, we could either use `rawEvents` property. But ideally the events should not be part of the Calendars file.  Instead, this should be the url of the gaia file. Therefore, we need a flag `data.encrypted`

Reply:
But for the verification, I propose to use 

`let calendar = {
  ...
  type: "ics-raw"
  data: {events: fileReader.result}
`